configure.ac

```diff
@@ -2027,7 +2030,10 @@ case "$CC_BASENAME" in
     case $ac_sys_system in
       Darwin*)
```

Comment:
BTW (and off-topic for this PR): this check should be for `cc_is_clang`, not `$ac_sys_system = Darwin`

Reply:
Will follow up in issue #96398.